hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -233,6 +233,10 @@ protected Dialect() {
 		registerHibernateType( Types.CLOB, StandardBasicTypes.CLOB.getName() );
 		registerHibernateType( Types.REAL, StandardBasicTypes.FLOAT.getName() );
 
+		if(supportsPartitionBy()) {
+			registerKeyword( "PARTITION" );
+		}
+
```

Comment:
I analyzed your suggestion, but to build an `IdentifierHelper`, I require access to `databaseMetaData` and to an `IdentifierHelperBuilder` which, in turn, required `JdbcEnvironment`. Unfortunately, we don't have those in the `DialectChecks` `match` contract. Even if we change the `match` method signature, how can we supply those through the current `CustomRunner`. Isn't it much more complex than the current implementation?